src/app.rs

```diff
@@ -290,51 +290,63 @@ fn validate_date_argument(arg: String) -> Result<(), String> {
 }
 
 pub fn validate_time_format(formatter: &str) -> Result<(), String> {
-    let mut chars = formatter.chars();
-    loop {
-        match chars.next() {
-            Some('%') => match chars.next() {
-                Some('.') => match chars.next() {
-                    Some('f') => (),
+    let str = formatter.to_string();
+    let vec: Vec<&str> = str.split('\n').collect();
```

Comment:
```suggestion
    let vec: Vec<&str> = formatter.split('\n').collect();
```
No need to create an additional owned string.